meilisearch/src/search.rs

```diff
@@ -1045,6 +1064,26 @@ fn make_hits(
         let mut document =
             permissive_json_pointer::select_values(&displayed_document, attributes_to_retrieve);
 
+        if retrieve_vectors {
+            let mut vectors = serde_json::Map::new();
+            for (name, mut vector) in index.embeddings(rtxn, id)? {
+                let user_defined = embedding_configs
+                    .iter()
+                    .find(|conf| conf.name == name)
+                    .is_some_and(|conf| conf.user_defined.contains(id));
+                let mut embedding = serde_json::Map::new();
+                embedding.insert("userDefined".to_string(), user_defined.into());
+                if vector.len() == 1 {
```

Comment:
rather than checking len and unwrap, you can probably match on the mut slice and take the vec here: see [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=99af8ab191d2743fea5c9c414662808a)

Reply:
It’s way better thank you!